pkg/infra/iac2/iac_templates.go

```diff
@@ -0,0 +1,121 @@
+package iac2
+
+import (
+	"context"
+	_ "embed"
+	"github.com/klothoplatform/klotho/pkg/core"
+	"github.com/klothoplatform/klotho/pkg/query"
+	sitter "github.com/smacker/go-tree-sitter"
+	"github.com/smacker/go-tree-sitter/typescript/typescript"
+	"io"
+	"regexp"
+	"strings"
+	"text/template"
+)
+
+type (
+	ResourceCreationSignature struct {
+		inputTypes map[string]string
+		outputType string
+	}
+
+	ResourceCreationTemplate struct {
+		ResourceCreationSignature
+		imports            map[string]struct{}
+		expressionTemplate string
+	}
+)
+
+var (
+	tsLanguage = core.SourceLanguage{
+		ID:     core.LanguageId("ts"),
+		Sitter: typescript.GetLanguage(),
+	}
+
+	parameterizeArgsRegex = regexp.MustCompile(`args(\.\w+)`)
+
+	//go:embed find_args.scm
+	findArgsQuery string
+
+	//go:embed find_create_func.scm
+	findCreateFuncQuery string
+
+	//go:embed find_imports.scm
+	findImportsQuery string
+)
+
+func ParseResourceCreationTemplate(contents []byte) ResourceCreationTemplate {
+	node := parseFile(contents)
+
+	result := ResourceCreationTemplate{}
+
+	// inputs
+	result.inputTypes = make(map[string]string)
+	nextInput := doQuery(node, findArgsQuery)
+	for {
+		match, found := nextInput()
+		if !found {
+			break
+		}
+		inputName, inputType := match["property_name"].Content(), match["property_type"].Content()
+		result.inputTypes[inputName] = inputType
+	}
+
+	// return type
+	returnTypeFunc := query.Select(doQuery(node, findCreateFuncQuery), query.ContentOf(query.ParamNamed("return_type")))
+	returnType, err := returnTypeFunc()
+	if !err {
+		// unexpected, since all inputs are from resources in the klotho binary
+		panic("couldn't find valid create() function")
+	}
+	result.outputType = returnType
+
+	// return expression
+	returnBodyFunc := query.Select(doQuery(node, findCreateFuncQuery), query.ContentOf(query.ParamNamed("return_body")))
```

Comment:
nit: can the `doQuery` results be extracted into its own variable so it's not queried twice (first time on L65).

Reply:
I tried it (after your comment), and it failed — the extracted var maintains state, do the second `Select` fails.

Since you gave the approval, I'll go ahead and merge as-is for now. But, open question: should I just remove these `query.Select`s altogether (if so, it'll probably be next week). I personally like them, but if I'm the only one (and I think I may be) then it's probably not worth the overall cognitive load.